### PR TITLE
HOT FIX: added support for many_many through relationship

### LINKED ISSUE #7: many_many through relationship joined on a separate DataObject fails
Hello,

I have this issue.

This module does not support many_many through relationship joined on a separate DataObject (https://docs.silverstripe.org/en/4/developer_guides/model/relations/#many-many-through-relationship-joined-on-a-separate-dataobject).

### PR DESCRIPTION
**Pull request by rasstislav**
Fixes #7

### CODE REVIEW

**Comment by gurucomkz** [comment]
It's a great feature!
Do you mind producing a test for it as well?

**Comment by rasstislav** [comment]
> It's a great feature! Do you mind producing a test for it as well?

Tests are my weakness :-/.

**Comment by gurucomkz** [comment]
How about a simplified use case - like what you have there in your project - minimal example of the things that must be replicated?
And we'll start from there, I'll help.

I'm sorry, but this repo is already used in some heavy projects and we must make sure the updates don't break stuff.

**Comment by gurucomkz** [comment]
Can you, please, rebase this update onto the `master` branch? It has updates to the phpunit which makes the tests work again.

**Comment by rasstislav** [comment]
> Can you, please, rebase this update onto the `master` branch? It has updates to the phpunit which makes the tests work again.

rebased :-)

File: src/EagerLoadedDataList.php [hunk 1]
```diff
@@ -213,12 +214,14 @@ public function eagerLoadManyMany(&$data, $manyManys, $withManyManys)
                 'map' => [],
             ];
 
+            $table = DataObject::getSchema()->tableName($component['join']);
```
Comment (gurucomkz) [inline src/EagerLoadedDataList.php]: This breaks existing functionality.
I suggest
```php
$table = isset($depClass['through'])
                ? DataObject::getSchema()->tableName($depClass['through'])
                : $component['join'];
```

Comment (rasstislav) [reply src/EagerLoadedDataList.php]: Fixed, thanks! Just tested with two standards many_many and two many_many through.

**Comment by rasstislav** [comment]
> How about a simplified use case - like what you have there in your project - minimal example of the things that must be replicated? And we'll start from there, I'll help.
> 
> I'm sorry, but this repo is already used in some heavy projects and we must make sure the updates don't break stuff.

I understand, but I have never ran SilverStripe tests. I don't know how to do it. Could you please do that?

**Comment by gurucomkz** [comment]
I think that my latest commit should be enough to get you running. Let me know if it works for you.

**Comment by rasstislav** [comment]
> I think that my latest commit should be enough to get you running. Let me know if it works for you.

Yes, it works, great! :-)

**Pull request merged**